packages/runtime/src/microapp-runtime.ts

```diff
@@ -177,4 +163,76 @@ export class MicroappRuntime {
 
     this.#iframe.dataset.hasRoutingScript = 'true';
   };
+
+  #handleIframeResize = ({ height }: { height: number }) => {
+    if (this.#iframe) {
+      this.#iframe.style.height = `${height}px`;
+      this.#iframe.style.minHeight = 'unset';
+    }
+  };
+
+  #injectIframeDimensionsScript = () => {
+    const iframeDoc = this.#iframe.contentDocument;
+    if (!iframeDoc) return;
+
+    if (this.#iframe.dataset.hasResizeScript) {
+      return;
+    }
+
+    const script = iframeDoc.createElement('script');
+    script.id = 'microapp-resize-script';
+    script.textContent = `
+      if (!window.__microappResizeInitialized) {
```

Comment:
My two cents: wrap this whole script into an anonymous function so that it doesn't conflict with the microapp's window:

```ts
(function() {
  // this is an anon function
}());
```